ios/RegionManager.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  RegionManager.swift
+//  Magnet
+//
+//  Created by Francisco Jordano on 09/11/2016.
+//  Copyright © 2016 Mozilla. All rights reserved.
+//
+
+import Foundation
+import CoreLocation
+import MagnetScannerIOS
+
+@objc(RegionManager) class RegionManager: NSObject, CLLocationManagerDelegate {
+  let locationManager:CLLocationManager = CLLocationManager()
+  private static let REGION_NAME = "region.magnet.mozilla.org"
+  var scanner: MagnetScanner? = nil
+  var isEnabled = true
+  var currentRegion: CLRegion?
+  let locationResolver: OneShotLocation = OneShotLocation()
+  
+  @objc override init() {
+    super.init()
+    
+    isEnabled = CLLocationManager.isMonitoringAvailableForClass(CLRegion.self)
+    
+    guard isEnabled else {
+      Log.w("Region monitoring not enabled for this device")
+      return;
+    }
+    
+    scanner = MagnetScanner(callback: self.onItemFound)
+  }
+  
+  private func onItemFound(item: Dictionary<String, AnyObject>) {
+    Log.l("Got item cause region abandoned \(item)")
```

Comment:
I find this wording confusing., don't think 'abandoned' is the right word. Perhaps: 'found item on leaving region'.